Enable Set/Reset (write 0x29 to Z_MSB)

hussam-qawaqzeh opened the pull request.

hussam-qawaqzeh committed:
Add files via upload

hussam-qawaqzeh committed:
Add 'QMC5883P' to device choices in MagDevice

hussam-qawaqzeh committed:
Add MAG_QMC5883P to MagDevice.h enum

hussam-qawaqzeh committed:
Add support for MagQMC5338P device detection

hussam-qawaqzeh committed:
Add QMC5883P to supported magnetometers list

hussam-qawaqzeh committed:
Update MagDevice.cpp

hussam-qawaqzeh committed:
BusDevice.cpp

hussam-qawaqzeh committed:
Update MagQMC5338P.h

hussam-qawaqzeh committed:
Update Hardware.cpp

hussam-qawaqzeh committed:
Merge branch 'rtlopez:master' into master

hussam-qawaqzeh committed:
Update MagQMC5338P.h

Enable Set/Reset (write 0x29 to Z_MSB)

hussam-qawaqzeh committed:
Update MagQMC5338P.h

hussam-qawaqzeh committed:
Update MagQMC5338L.h

hussam-qawaqzeh committed:
Update Hardware.cpp

hussam-qawaqzeh committed:
Update MagQMC5338L.h

hussam-qawaqzeh committed:
Update MagQMC5338P.h

rtlopez commented on lib/Espfc/src/Device/MagQMC5338P.h:
```diff
@@ -45,18 +41,22 @@ class MagQMC5338P : public MagDevice
-        // We use ±8G (0x02) as the default range for the drone.
-        _currentRange = QMC5883P_RANGE_8G;
-        setMode(_currentRange);
+        // 🔑 Step 1: Enable Set/Reset (write 0x29 to Z_MSB)
```
Plase remove UTF symbols from code.

rtlopez reviewed (commented):
1. what problem does this change solve?
2. there is a typo in file name, QMC5338P but should be QMC5883P.

hussam-qawaqzeh committed:
remove UTF symbols from code

hussam-qawaqzeh committed:
Update MagQMC5338P.h

The improvements I added to the QMC5883P driver in MagQMC5338P.h are:

Aligned the driver more closely with the Adafruit library: same I2C address (0x2C), same chip ID (0x80), and the same CONTROL1 / CONTROL2 register layout.
Improved sensor detection in testConnection(): it now retries multiple times instead of relying on a single read, which helps during startup.
Improved initialization in begin(): I added softReset(), then explicitly configured the range, set/reset mode, and continuous measurement mode at 100Hz.
Made initialization safer: any failed write or read now causes a real initialization failure instead of continuing as if the sensor was detected successfully.
Improved compatibility with the project architecture: periodic reading now uses readMag() with readFast() like the other magnetometer drivers, and includes an initial read to support BusSlave.
Improved raw-to-Gauss conversion in convert() based on the active range instead of relying on a weaker implicit assumption.
Cleaned up the implementation by replacing magic numbers with explicit sensor constants, which makes the driver easier to maintain.
